src/resmom/mom_main.c

```diff
@@ -8989,38 +8905,38 @@ main(int argc, char *argv[])
 			log_err(errno, msg_daemonname, "setsid failed");
 			return (2);
 		}
-		mom_lock(lockfds, F_WRLCK);	/* lock out other MOMs */
+		if (lock_file(lockfds, F_WRLCK, "mom.lock", 1, NULL, 0)) {		/* lock out other MOMs */
+			log_errf(errno, msg_daemonname, "pbs_mom: another mom running");
+			fprintf(stderr, "%s\n", "pbs_mom: another mom running");
+			exit(1);
+		}
 	}
-	mom_pid = getpid();
-
-	(void)fclose(stdin);
-	(void)fclose(stdout);
-	(void)fclose(stderr);
-	dummyfile = fopen("/dev/null", "r");
+#endif		
+	fclose(stdin);
+	fclose(stdout);
+	fclose(stderr);
+	dummyfile = fopen(NULL_DEVICE, "r");
 	assert((dummyfile != 0) && (fileno(dummyfile) == 0));
-	dummyfile = fopen("/dev/null", "w");
+	dummyfile = fopen(NULL_DEVICE, "w");
 	assert((dummyfile != 0) && (fileno(dummyfile) == 1));
-	dummyfile = fopen("/dev/null", "w");
+	dummyfile = fopen(NULL_DEVICE, "w");
 	assert((dummyfile != 0) && (fileno(dummyfile) == 2));
 #else	/* DEBUG */
 	if (stalone != 1) {
-		(void) sprintf(log_buffer, "Debug build does not fork.");
+		sprintf(log_buffer, "Debug build does not fork.");
```

Comment:
since there is not string formatting in this message you can pass msg directly to log_record